### PR TITLE
docs: add CORS documentation to 'Configuration Overview' page

### PR DESCRIPTION
**Pull request by hassankhan**
**Check List**
- [x] Tests has been run in packages where changes made if available
- [x] Linter has been run for changed code
- [x] Tests for the changes have been added if not covered yet
- [x] Docs have been added / updated if required

**Description of Changes Made (if issue reference is not provided)**

Add documentation describing how to configure CORS for the REST API.

### CODE REVIEW

**Comment by orbit88** [comment]
Better to keep it my way to  secure my privacy. Also avoid exploit my personal belongings

**Comment by hassankhan** [comment]
Hi @orbit88, could you clarify what you mean? This PR is only a documentation-related change, it doesn't change any Cube.js functionality at all.

File: docs/Configuration/Overview.md [hunk 1]
```diff
@@ -5,8 +5,8 @@ category: Configuration
 menuOrder: 1
 ---
 
-Cube.js is designed to work with different configuration sources. There
-are two main ways you can set configuration options; via [a configuration
+Cube.js is designed to work with different configuration sources. There are two
+main ways you can set configuration options; via [a configuration
```
Comment (igorlukanin) [inline docs/Configuration/Overview.md:9]:
```suggestion
ways you can set configuration options; via [a configuration
```

"Main" is redundant here, right?

File: docs/Configuration/Overview.md [hunk 2]
```diff
@@ -15,30 +15,54 @@ variables][link-env-vars].
 
 ## Configuration Precedence
 
-In Cube.js, environment variables take precedence over values specified
-in `cube.js`. In production scenarios, it allows changing configuration
-by simply switching environment variables, which is generally safer,
-faster and easier than re-deploying the codebase.
+In Cube.js, environment variables take precedence over values specified in
+`cube.js`. In production scenarios, it allows changing configuration by simply
```
Comment (igorlukanin) [inline docs/Configuration/Overview.md:19]:
```suggestion
`cube.js`. In production scenarios, it allows changing configuration by
```

How simple it is, depends on the tools used, right? But won't it be more concise to omit "simply" here?

File: docs/Configuration/Overview.md [hunk 3]
```diff
@@ -15,30 +15,54 @@ variables][link-env-vars].
...
-- Logs incorrect/invalid configuration for `externalRefresh` /`waitForRenew` instead of throwing errors
+- Logs incorrect/invalid configuration for `externalRefresh` /`waitForRenew`
+  instead of throwing errors
 
-## Migration from Express to Docker template
+## Configuring CORS
+
+The Cube.js REST API supports Cross-Origin Resource Sharing (CORS) through the
+[Express CORS middleware][link-express-cors]. The middleware can be configured
+by setting [any available options][link-express-cors-opts] in the `cube.js`
```
Comment (igorlukanin) [inline docs/Configuration/Overview.md:40]:
```suggestion
by setting [relevant options][link-express-cors-opts] in the `cube.js`
```

Maybe saying "any" is quite misleading here?

File: docs/Configuration/Overview.md [hunk 4]
```diff
@@ -15,30 +15,54 @@ variables][link-env-vars].
...
+
+## Migrating from Express to Docker
 
 Since [`v0.23`][link-v-023-release], Cube.js CLI uses the `docker` template
 instead of `express` as a default for project creation, and it is the
-recommended route for production. To migrate you should move all of your
-Cube.js dependencies in `package.json` to `devDependencies` and leave all of
-your dependencies that you use to configure Cube.js in `dependencies`.
+recommended route for production. To migrate you should move all of your Cube.js
+dependencies in `package.json` to `devDependencies` and leave all of your
```
Comment (igorlukanin) [inline docs/Configuration/Overview.md:64]:
```suggestion
recommended route for production. To migrate you should move all Cube.js
dependencies in `package.json` to `devDependencies` and leave
```

Just curious, is it correct from the language standpoint to remove these?

**Review by igorlukanin — APPROVED** [review]

**Pull request merged**